spec/controllers/admin/users_spec.rb

```diff
@@ -0,0 +1,88 @@
+require "rails_helper"
+
+RSpec.describe Admin::UsersController, type: :controller do
+  before(:each) do
+    admin = FactoryGirl.create(:user, admin: true)
+    allow_message_expectations_on_nil
+    allow(request.env["warden"]).to receive(:authenticate!) { admin }
+    allow(controller).to receive(:current_user) { admin }
+  end
+
+  describe "#index" do
+    before(:each) do
+      get :index
+    end
+
+    it "Is successful" do
+      expect(response).to be_success
+    end
+
+    it "Returns renders template #index" do
+      expect(response).to render_template("index")
+    end
+  end
+
+  describe "#edit" do
+    before(:each) do
+      user = FactoryGirl.create(:user)
+      get :edit, id: user.id
+    end
+
+    it "Is successful" do
+      expect(response).to be_success
+    end
+
+    it "Returns renders template #edit" do
+      expect(response).to render_template("edit")
+    end
+  end
+
+  describe "#update with valid attributes" do
+    before(:each) do
+      user = FactoryGirl.create(:user)
+      put :update, { id: user, user: FactoryGirl.attributes_for(:user, :username => "Test")}
```

Comment:
Line is too long. [92/80]<br>Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside } missing.